### PR TITLE
feat: Allow viewing content generated by AINSERT instruction and preprocessors

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: language_server/test/virtual_file_provider_test.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Broadcom.
```
Comment (jirimosinger) [inline language_server/test/virtual_file_provider_test.cpp:2]: 2022

File: parser_library/src/processing/opencode_provider.cpp [hunk 2]
```diff
@@ -338,7 +362,8 @@ bool opencode_provider::try_running_preprocessor()
 
     auto virtual_copy_name = m_ctx->hlasm_ctx->ids().add("preprocessor:" + std::to_string(current_line));
```
Comment (jirimosinger) [inline parser_library/src/processing/opencode_provider.cpp:363]:
```suggestion
    auto virtual_file_name = m_ctx->hlasm_ctx->ids().add("preprocessor:" + std::to_string(current_line));
```

File: parser_library/src/processing/opencode_provider.cpp [hunk 3]
```diff
@@ -419,11 +446,13 @@ void opencode_provider::convert_ainsert_buffer_to_copybook()
     auto virtual_copy_name =
         m_ctx->hlasm_ctx->ids().add("AINSERT:" + std::to_string(m_ctx->hlasm_ctx->obtain_ainsert_id()));
 
-    auto new_file = m_virtual_files.try_emplace(virtual_copy_name, std::move(result)).first;
+    auto new_file = m_virtual_files.try_emplace(virtual_copy_name, std::move(result), next_virtual_file_id()).first;
```
Comment (jirimosinger) [inline parser_library/src/processing/opencode_provider.cpp:449]: It feels a little bit weird to deal with the virtual file IDs. Consider moving this responsibility somewhere else.

**Review by jirimosinger — COMMENTED** [review]
Consider adding some tests that would verify handling of virtual files in VSCode

File: parser_library/src/processing/opencode_provider.cpp (hunk 3 above)
Comment (slavek-kucera) [reply parser_library/src/processing/opencode_provider.cpp:449]: I've actually realized that there is a much bigger (lifetime) issue, so I will rework this.

**Review by jirimosinger — APPROVED** [review]

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**